website/docs/reference/widgets/menu-button.md

```diff
@@ -1,76 +1,89 @@
-| **Menu Variant**   | Sets the variant of the menu button.                   |
-| **Border Radius**  | Allows you to define curved corners.                   |
-| **Box Shadow**     | Allows you to choose from the available shadow styles. |
-| **Icon**           | Sets the icon to be used for menu button.              |
-| **Placement**      | Sets the space between items.                          |
-| **Icon Alignment** | Sets the alignment of the selected icon.               |
+##### Menu item settings
 
-### Menu Item Settings
+The gear icon allows you to access and configure settings for a menu item, such as its [properties](#menu-item-properties). This can be useful for customizing the behavior and appearance of the menu item. You can also set [styles specific to the menu item](#menu-item-properties) like Icon, the position of the icon, background color, and more.
 
-Each menu item can now be customized through a set of properties by clicking on the gear icon.
+To specify an action to be performed when a menu item is clicked, you can bind a supported action to the `onClick` event of the menu item. You can define different actions to be triggered for different menu items using the gear icon next to them. For more information, see the list of [supported actions](/reference/appsmith-framework/widget-actions)
 
-![](/img/Menu_items.gif)
+### Dynamic
+When you select the dynamic source, you can bind the menu item's property to a data source, such as a database query or API response. This allows the widget to dynamically populate the menu items based on available data. This is useful for creating menus that can change and adapt based on user input or application state. You can configure the menu items by clicking the `Item Configuration` button.
 
-| Property                           | Description                                                                     |
-| -----------------------------------| -------------------------------------------------------------------------------------------------------------|
-| **label**                          | Sets the label of a menu item.          |
-| **Visible**                        | Controls the visibility of a menu item. |
-| **Disabled**                       | Disables input to a menu item.          |
+:::note
+ Dynamic menu items are **only** supported inside the menu button and the list widget. You can add up to 10 menu items by defining them in the "Source Data" property for a dynamic source.
+:::
 
-| Icon Options       | Description                               |
-| ------------------ | ----------------------------------------- |
-| **Icon**           | Sets the icon to be used for menu button. |
-| **Icon alignment** | Sets the alignment of the selected icon.  |
+If the dynamic source array has more than 10 items, the `SourceData` fields show an error message: `Source data cannot have more than 10 items`.
```

Comment:
this should be in the troubleshooting guide with the error message. no need to add it here, as we have already mentioned it in the callout. Is that ok @dilippitchika